### PR TITLE
fixed bugs in conv_operator; add calc_output_size

### PR DESCRIPTION
**Pull request by yu239**

### CODE REVIEW

File: python/paddle/trainer/config_parser.py [hunk 1]
```diff
@@ -686,12 +685,15 @@ def __init__(
         if num_filters is not None:
             self.operator_conf.num_filters = num_filters
 
-        parse_conv(conv_conf, input_layer_names[0], self.operator_conf.conv_conf, True)
+        parse_conv(conv_conf,
+                   MakeLayerNameInSubmodel(input_layer_names[0]),
```
Comment (emailweixu) [inline python/paddle/trainer/config_parser.py]: MakeLayerNameInSubmodel should be handled at the base class Operator

File: python/paddle/trainer_config_helpers/layers.py [hunk 2]
```diff
@@ -2668,7 +2668,7 @@ def __add_evaluator__(e):
     return LayerOutput(name, LayerType.COST, parents=[input, label])
 
 def conv_operator(input, filter_size, num_filters,
-                  num_channel=None, stride=1, padding=0,
+                  num_channel=None, stride=1, padding=0, groups=1,
```
Comment (emailweixu) [inline python/paddle/trainer_config_helpers/layers.py]: suggest to split input into two parameters: image and kernel to make the interface clearer

File: python/paddle/trainer/config_parser.py (hunk 1 above)
Comment (yu239) [inline python/paddle/trainer/config_parser.py]: Only this place is needed. If you add MakeLayerNameInSubmodel to the base class, the Operator would be fine, but when feeding the operator to the input of the MixedLayer, the parent model name will be added twice.

Comment (emailweixu) [inline python/paddle/trainer/config_parser.py]: I see.

**Pull request merged**